### ISSUE TITLE
Issues importing to Security Hub when a finding has a lot of results

### ISSUE THREAD

**Issue by Kirizan** [toniblyx/prowler#950]
While running a Prowler Scan in an account with 469 Lambda functions and two Cloud Trails, generating 938 findings for [extra720](https://github.com/toniblyx/prowler/blob/42e54c42cf878eb3928465c981f838765062eb01/checks/check_extra720).

The error message given is:

    prowler/include/securityhub_integration: line 53: /usr/bin/jq: Argument list too long

This line in [securityhub_integration](https://github.com/toniblyx/prowler/blob/42e54c42cf878eb3928465c981f838765062eb01/include/securityhub_integration) is trying to get a list of the prior findings. I believe the error is happening at `--argjson ids "$NEW_FINDING_IDS"` part of the command.

I modified the line to replace the double quotes with single quotes for that part so it looks like `--argjson ids '$NEW_FINDING_IDS'`. I did it based on [this](https://stackoverflow.com/questions/11475221/why-do-i-get-bin-sh-argument-list-too-long-when-passing-quoted-arguments) Stack Overflow question.

I'm guessing this issue will also appear whenever there are a lot of findings. I'm not sure if there is a good way to test the limits of how many findings can be sent at a time, and break them down so Prowler can send in batches based on how many can be sent at once.

**Comment by Kirizan** [comment]
Changing the double quotes to single quotes didn't actually fix this issue.

**Comment by Kirizan** [comment]
To replicate this issues, do the following (directions applied from [here](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-awscli.html)):

First, create a role for the Lambda functions:

    aws iam create-role --role-name lambda-ex --assume-role-policy-document '{"Version": "2012-10-17","Statement": [{ "Effect": "Allow", "Principal": {"Service": "lambda.amazonaws.com"}, "Action": "sts:AssumeRole"}]}'

Copy the role arn from the output

Then attach the execution role:

    aws iam attach-role-policy --role-name lambda-ex --policy-arn arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole

create a file called `index.js` with the following contents: 
```
exports.handler = async function(event, context) {
  console.log("ENVIRONMENT VARIABLES\n" + JSON.stringify(process.env, null, 2))
  console.log("EVENT\n" + JSON.stringify(event, null, 2))
  return context.logStreamName
}
```

Zip the file:

    zip function.zip index.js

Run the following command:
```
for i in {001..500}; do aws lambda create-function --function-name my-function-${i} --zip-file fileb://function.zip --handler index.handler --runtime nodejs12.x --role <ARN FROM ABOVE>; done
```

This will create 500 Lambda functions. Next, ensure you have at least 2 cloud-trails in the region you created the above functions in.

Finally, run Prowler and add the flag `-M json-asff -S -c extra720` to the prowler command, so it attempts to send results to Security Hub. It will take around 30 minutes to finish the check, which will result in the above errors.

**Comment by toniblyx** [comment]
Thanks @Kirizan, good finding, I wonder if using [server-side pagination](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-serverside) may make the trick. By default AWS CLI uses page size of 1000, I have used max-items option in some checks like in `check_extra741` then, usign `-m <maxitems>` option that can be managed. I haven't used `--starting-token` so far but I think it is part of the solution here. Busy with other bug now if you can give it a look to that possible solution with starting token let me know.

**Comment by Kirizan** [comment]
@toniblyx This issue doesn't appear to be the securityhub command (for now), but it's after the pipe in the `jq` command. It's trying to pass 1542 findings to the `--argjson ids "$NEW_FINDING_IDS` section on line 53, which combined with the rest of the command is making the `jq` command to long to process. I also think this would happen regardless of which check was being ran, if it has 1000's of findings, this will happen.

I think the service-side pagination might be needed when the 1542 findings are in security hub, but for now I don't think that's the problem. At this point in the scan, there are no findings in Security Hub for the command to return, so it's not close to the 1000 in the default page.

I'm trying to think of a way to get the two lists (existing and new) and then process the lists separately, then send the results to security hub.

**Comment by toniblyx** [comment]
What I mean is, if we control or reduce the number of findings showed from `securityhub get-findings` we could avoid the jq issue, couldn't we?

**Comment by Kirizan** [comment]
I was thinking the issue was from having 1542 findings passed to the JQ command from what Prowler found, I wasn't thinking that `securityhub get-findings` was also getting 1542 results, meaning JQ is trying to process 3084 items. We still might need to control both results however, I'll do some testing and see what I find.

**Comment by Kirizan** [comment]
@toniblyx I just modified the `securityhub get-findings` command to max out at 100 results, and still got the same error. The 1542 findings being sent to the `argjson` flag is still to much. I'll play with the code some and see what can be done to solve this.

**Comment by toniblyx** [comment]
Closing the issue since it was fixed in PR #953. Thanks @Kirizan
